src/main/java/org/wso2/carbon/connector/utils/EmailParser.java

```diff
@@ -0,0 +1,115 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.connector.utils;
+
+import org.apache.commons.mail.util.MimeMessageParser;
+import org.wso2.carbon.connector.exception.EmailParsingException;
+import org.wso2.carbon.connector.pojo.Attachment;
+import org.wso2.carbon.connector.pojo.EmailMessage;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.activation.DataSource;
+import javax.mail.Message;
+import javax.mail.internet.MimeMessage;
+
+import static java.lang.String.format;
+
+/**
+ * Parses emails
+ */
+public final class EmailParser {
+
+    private EmailParser() {
+
+    }
+
+    /**
+     * Gets email content and attachments
+     *
+     * @param messages List of messages to be parsed
+     * @return Parsed messages
+     * @throws EmailParsingException if failed to parse content
+     */
+    public static List<EmailMessage> parseMessageList(List<Message> messages) throws EmailParsingException {
```

Comment:
Is this method used multiple times? If not, we can make this a private method of the referred class.

Reply:
moved to the class level